shared.c

```diff
@@ -975,3 +989,356 @@ int appimage_unregister_in_system(char *path, gboolean verbose)
     g_free(md5);
     return 0;
 }
+
+
+int match_regexp(const gchar *text, const gchar *pattern)
```

Comment:
That seems like a good idea, we could even narrow it further to `const char const* text, const gchar const* pattern`.